docs/apis/subsystems/output/index.md

```diff
@@ -198,66 +198,76 @@ For each of the functions below we'll try to explain when they should be used, e
 
 ### String formatting functions
 
+The format_string and format_text functions should always be used when preparing the output of information. They may also be used to process information before it is stored in the database however, filters should only be applied at output. For example, language filters must only be applied as the content is prepared for output because we don't yet know the user's preferred language.
+
 #### p() and s()
 
 ```php
 function s($var, $strip=false)
 function p($var, $strip=false)
 ```
 
-These functions share the same code, so they will be explained together. The only difference is that `s()` returns the string, while `p()` prints it directly.
+The only difference between these two functions is that `s()` returns the string, while `p()` prints it directly.
 
 These functions should be used to:
 
 - print all the **values of form fields** like `<input>` or `<textarea>` tags.
 - **show plain (non-HTML) text** that has been introduced by the user (search string, quiz responses, ...).
-- print, in general, all the **dynamic data, not being HTML**, that doesn't need to be cleaned nor processed by filters. It is important not to use these functions for strings that contain HTML markup.
+- print, in general, all **dynamic data that is not HTML**, does not need to be cleaned or processed by filters. Never use these functions for strings that contain HTML markup.
```

Comment:
I wonder if we can rewrite this to be clearer. The double negative is not easy to parse.

```suggestion
- print, in general, all **dynamic HTML data must be cleaned or processed by filters**. Data that _can not_ contain HTML markup does not need to be processed. Never use these functions for strings that _might_ contain HTML markup.
```

It's also worth highlighting that it's anything that could possibyl contain HTML, even if that wasn't the intent. Anything whose PARAM_ type includes possible HTML.